### ISSUE TITLE
Either support or disallow wildcard params not at the end of path

### ISSUE THREAD

**Issue by aeriksson** [metosin/reitit#568]
Currently, Reitit allows users to create routes with paths on the form `/.../{*param}/...`.

However, routing doesn't seem to work for such routes (it seems to give 'false' matches, and path parameters aren't extracted properly).

Either such routes should work, or Reitit should provide an error/warning when they are created.

**Comment by ikitommi** [comment]
Good catch, this should be fixed (e.g. not allow them in non-tail postion)!
